### PR TITLE
SPSH-409 False Return Code (500 instead of 404)

### PR DESCRIPTION
**Pull request by casparneumann-cap**
# Description
/api/organisationen/{id}/administriert AND /api/organisationen/{id}/zugehoerig both returned a 500 Status Code for an unknown organisation id, instead of correctly returning a 404.

This has been fixed so that now a 404 is returned.

## Links to Tickets or other pull requests
https://ticketsystem.dbildungscloud.de/browse/SPSH-409

## Approval for review
- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

### CODE REVIEW

File: src/modules/organisation/api/organisation.controller.spec.ts [hunk 1]
```diff
@@ -261,41 +261,52 @@ describe('OrganisationController', () => {
         const params: OrganisationByIdParams = {
             organisationId: faker.string.uuid(),
         };
+        describe('when usecase returns a OrganisationResponse', () => {
```
Comment (marode-cap) [inline src/modules/organisation/api/organisation.controller.spec.ts:264]: I know we're not totally consistent, but a few empty lines would aid readability (also in line 301)

```suggestion
        };

        describe('when usecase returns a OrganisationResponse', () => {
```

**Review by marode-cap — CHANGES_REQUESTED** [review]

File: src/modules/organisation/api/organisation.uc.spec.ts [hunk 2]
```diff
@@ -257,16 +257,19 @@ describe('OrganisationUc', () => {
...
+                if (!(result instanceof SchulConnexError)) {
+                    expect(result.total).toBe(2);
+                    expect(result.items).toHaveLength(2);
+                    expect(result.items[0]?.name).toEqual(organisationDos[0]?.name);
+                    expect(result.items[1]?.name).toEqual(organisationDos[1]?.name);
+                    expect(result.items[0]?.kennung).toEqual(organisationDos[0]?.kennung);
+                    expect(result.items[1]?.kennung).toEqual(organisationDos[1]?.kennung);
+                    expect(result.items[0]?.typ).toEqual(organisationDos[0]?.typ);
+                    expect(result.items[1]?.typ).toEqual(organisationDos[1]?.typ);
+                }
```
[the remark this answers is not in the document]
Comment (casparneumann-cap) [reply src/modules/organisation/api/organisation.uc.spec.ts:272]: fixed

File: src/modules/organisation/api/organisation.uc.spec.ts [hunk 3]
```diff
@@ -303,16 +306,19 @@ describe('OrganisationUc', () => {
...
```
[the remark this answers is not in the document]
Comment (casparneumann-cap) [reply src/modules/organisation/api/organisation.uc.spec.ts:321]: fixed

File: src/modules/organisation/api/organisation.controller.spec.ts (hunk 1 above)
Comment (casparneumann-cap) [reply src/modules/organisation/api/organisation.controller.spec.ts:264]: fixed

File: src/modules/organisation/api/organisation.controller.ts [hunk 4]
```diff
@@ -193,8 +199,13 @@ export class OrganisationController {
...
-        const response: PagedResponse<OrganisationResponse> = new PagedResponse(organisations);
+        const result: Paged<OrganisationResponse> | SchulConnexError = await this.uc.findZugehoerigZu(
+            params.organisationId,
+        );
+        if (result instanceof SchulConnexError) {
+            throw SchulConnexErrorMapper.mapSchulConnexErrorToHttpException(result);
+        }
+        const response: PagedResponse<OrganisationResponse> = new PagedResponse(result);
 
         return response;
```
[the remark this answers is not in the document]
Comment (casparneumann-cap) [reply src/modules/organisation/api/organisation.controller.ts:210]: fixed

**Review by marode-cap — APPROVED** [review]

**Pull request merged**